app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt

```diff
@@ -142,52 +135,38 @@ fun ImportFilesScreen(
         transferOptionsCallbacks = transferOptionsCallbacks,
         addFiles = importFilesViewModel::importFiles,
         closeActivity = closeActivity,
-        integrityCheckResult = { integrityCheckResult },
+        sendStatus = { sendStatus },
         sendTransfer = importFilesViewModel::sendTransfer,
         shouldStartByPromptingUserForFiles = true,
-        isTransferStarted = { sendActionResult != SendActionResult.NotStarted },
         snackbarHostState = snackbarHostState,
     )
 }
 
 @Composable
 private fun HandleSendActionResult(
     snackbarHostState: SnackbarHostState,
-    getSendActionResult: () -> SendActionResult?,
+    sendStatus: () -> SendStatus,
     transferType: () -> TransferTypeUi,
     navigateToUploadProgress: (transferType: TransferTypeUi, totalSize: Long) -> Unit,
     resetSendActionResult: () -> Unit,
 ) {
-    val errorMessage = stringResource(R.string.errorUnknown)
-    LaunchedEffect(getSendActionResult()) {
-        when (val actionResult = getSendActionResult()) {
-            is SendActionResult.Success -> navigateToUploadProgress(transferType(), actionResult.totalSize)
-            is SendActionResult.Failure -> {
-                snackbarHostState.showSnackbar(errorMessage)
+    val context = LocalContext.current
+
+    LaunchedEffect(sendStatus()) {
+        when (val actionResult = sendStatus()) {
+            is SendStatus.Success -> navigateToUploadProgress(transferType(), actionResult.totalSize)
+            is SendStatus.Refused -> {
+                snackbarHostState.showSnackbar(context.getString(R.string.errorAppIntegrity))
```

Comment:
Shouldn't we also reset the status here ?

Reply:
Also for success in fact
When we reach one of the 3 final stages, we reset it, that seems logic

Reply:
Yes I'm afraid we can't avoid resetting the state